### PR TITLE
[RFC4_dev] Address code review comments

### PR DESCRIPTION
**Pull request by rouault**
Continuation of PR #1817. Commits specific to this one start at "Grid refactoring: address review comments of #1777 "

### CODE REVIEW

**Review by mwtoews — COMMENTED** [review]
Just a few minor corrections to docs to address.

File: docs/source/usage/environmentvars.rst [hunk 1]
```diff
@@ -56,3 +56,21 @@ done by setting the variable with no content::
...
+    (Content Delivery Network) storage, when grids are not available locally.
+    Alternatively, the :c:func:`proj_context_set_enable_network` function can
+    be used.
+
+.. envvar:: PROJ_NETWORK_ENDPOINT
+
+    .. versionadded:: 7.0.0
+
+    Define the endpoint of the CDN storage. Normally defined through the proj.ini
+    configuration file locale in PROJ_LIB.
```
[the remark this answers is not in the document]
Comment (kbevers) [reply docs/source/usage/environmentvars.rst]: Preferably the last option, it should link to the definition of PROJ_LIB elsewhere in the docs

**Comment by rouault** [comment]
> Just a few minor corrections to docs to address.

Thanks. Done per 8821b3591ea6598e764b1b749749e424a52feaf2

**Comment by rouault** [comment]
Any objection to merge this ?

**Comment by kbevers** [comment]
> Any objection to merge this ?

No. In this PR you've taken care of all the things we didn't like about the earlier PR's, so I think this should be fine to merge without further reviews :-)

**Comment by rouault** [comment]
Actually the title of the PR is a bit misleading. It started with code review-only changes, but commits starting with "pj_bilinear_interpolation_three_samples(): fix MSVC warning" are about other fixes & additions

**Comment by kbevers** [comment]
Okay, I see if I can find the time to look at those commits during the weekend. It'd be nice to get RFC4 stuff fully merge soon.

File: docs/source/resource_files.rst [hunk 2]
```diff
@@ -23,21 +23,34 @@ The following paths are checked in order:
...
   the :c:func:`proj_context_set_file_finder`. If it is set, the next tests
   will not be run.
+
 - Path(s) set with the :c:func:`proj_context_set_search_paths`. If set, the
   next tests will not be run.
+
+- The PROJ user writable directory, which is :
+
+    * on Windows, ${LOCALAPPDATA}/proj
+    * on MacOSX, ${HOME}/Library/Logs/proj
```
Comment (kbevers) [inline docs/source/resource_files.rst:37]: Is this really the correct plase to put files on OS X? On my OS X installation I see that QGIS has put files in `~/Library/Application Support/` and `~/Library/Caches/`. Both sounds like better places for us to put data. I couldn't easily find an overview of what is intended to go in the subfolders of `~/Library/`

**Review by kbevers — COMMENTED** [review]
Overall this looks good to me. I just need clarification on the OS X user writable directory and I am happy to merge this

File: docs/source/resource_files.rst (hunk 2 above)
Comment (rouault) [reply docs/source/resource_files.rst:37]: You're right. I was pointed during RFC-4 review to https://github.com/ActiveState/appdirs/blob/a54ea98feed0a7593475b94de3a359e9e1fe8fdb/appdirs.py#L45-L97 which indeed points to ~/Library/Application Support/, but in this file there was also uses of ~/Library/Logs and I erroneously used that. I'll change that to ~/Library/Application Support/ which makes more sense

Comment (rouault) [reply docs/source/resource_files.rst:37]: Fixed per 9d2851ad362a32e756986be65e04914230bd9ab5

Comment (rouault) [reply docs/source/resource_files.rst:37]: and doc update per 9bb87ea4782b6005848afde29db9751e05ee0e55

Comment (kbevers) [reply docs/source/resource_files.rst:37]: Thanks

**Pull request merged**